### PR TITLE
Fix backend e2e tests

### PR DESCRIPTION
**Pull request by anders-schneider**
Addresses #173 
- [X] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Adding pagination caused some backend e2e tests to break; this change fixes those.

Separately, I discovered that in `backend/core`, running `yarn test:e2e` is not idempotent because it expects the DB to only had seed data, but then it modifies/adds data and doesn't clean up after itself. I added a small cleanup routine to `listings.e2e-spec.ts`, but other tests are harder (e.g. the user e2e test). As a result, I changed the top-level `yarn test:e2e:backend:core` to run `yarn test:e2e:local` (which, in contrast to `yarn test:e2e`, creates the DB fresh before running its tests).

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

```
$ yarn test:e2e:backend:core
```

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have updated the changelog to include a description of my changes

### CODE REVIEW

File: backend/core/package.json [hunk 1]
```diff
@@ -126,4 +126,4 @@
     "node": "12",
     "yarn": "^1.22"
   }
-}
```
Comment (willrlin) [inline backend/core/package.json:129]: Do you have an editor that removes the newline at the end of JSON files?

File: backend/core/test/assets/assets.e2e-spec.ts [hunk 2]
```diff
@@ -70,7 +70,7 @@ describe("AssetsController", () => {
   describe("retrieve", () => {
     it("should return a paginated assets list", async () => {
       const res = await supertest(app.getHttpServer())
-        .get(`/assets/`)
+        .get(`/assets?page=1&limit=10`)
```
Comment (willrlin) [inline backend/core/test/assets/assets.e2e-spec.ts:73]: What makes this request need a page and limit?

File: backend/core/package.json (hunk 1 above)
Comment (anders-schneider) [reply backend/core/package.json]: Haha what was your first clue? ;)

Fixed it! (Both my VSCode settings, and re-added the newlines)

File: backend/core/test/assets/assets.e2e-spec.ts (hunk 2 above)
Comment (anders-schneider) [reply backend/core/test/assets/assets.e2e-spec.ts]: Good question! And it made me realize that what I actually wanted to do was make `GET /assets` resilient to the case where we set `page` and/or `limit` to -1. So I reverted the change to the test and instead change `assets.service.ts` to look like the applications service: https://github.com/CityOfDetroit/bloom/blob/da16a46128e20c1ebdeb999e2edfe41a81f22d97/backend/core/src/applications/applications.service.ts#L51

**Review by willrlin — APPROVED** [review]

**Pull request merged**